### PR TITLE
Another rounds of coop fixes for the runtime.

### PR DESCRIPTION
**Pull request by kumpera**
Now we're down to appdomain related crashes only.

### CODE REVIEW

File: mono/metadata/object.c [hunk 1]
```diff
@@ -6279,6 +6279,9 @@ mono_print_unhandled_exception (MonoObject *exc)
 	if (exc == (MonoObject*)mono_object_domain (exc)->out_of_memory_ex) {
 		message = g_strdup ("OutOfMemoryException");
 		free_message = TRUE;
+	} else if (exc == (MonoObject*)mono_object_domain (exc)->stack_overflow_ex) {
+		message = g_strdup ("StackOverflowException"); //if we OVF, we can't expect to have space to JIT the needed space.
```
Comment (alexrp) [inline mono/metadata/object.c]: I don't understand what this comment is referring to?

Comment (kumpera) [inline mono/metadata/object.c]: Oh, it's about having space to JIT the ToString method itself. The message is wrong.

**Comment by kumpera** [comment]
build

**Comment by alexrp** [comment]
Apart from the comment that needs clarification, this LGTM.

**Pull request closed without merging**